core/matrix/fbcsr.cpp

```diff
@@ -197,17 +198,13 @@ template <typename ValueType, typename IndexType>
 void Fbcsr<ValueType, IndexType>::convert_to(
     SparsityCsr<ValueType, IndexType>* const result) const
 {
-    auto exec = this->get_executor();
-    auto tmp = SparsityCsr<ValueType, IndexType>::create(
-        exec,
+    result->set_size(
         gko::dim<2>{static_cast<size_type>(this->get_num_block_rows()),
-                    static_cast<size_type>(this->get_num_block_cols())},
-        this->get_num_stored_blocks());
-
-    tmp->col_idxs_ = this->col_idxs_;
-    tmp->row_ptrs_ = this->row_ptrs_;
-    tmp->value_ = Array<ValueType>(exec, {one<ValueType>()});
-    tmp->move_to(result);
+                    static_cast<size_type>(this->get_num_block_cols())});
+    result->col_idxs_ = this->col_idxs_;
+    result->row_ptrs_ = this->row_ptrs_;
+    result->value_ =
+        Array<ValueType>(result->get_executor(), {one<ValueType>()});
```

Comment:
Yeah, this conversion is not like the others, it converts a Fbcsr matrix into its block sparsity pattern, which is not ideal any maybe not what we want? But the interface promises it, so I needed to follow it.

Reply:
Does any fbcsr test generator uses this conversion? or we expand the col/row according to the block size.

Reply:
But the interface expects conversion to SparsityCsr and not to SparsitBlockCsr, right ? I would maybe prefer if we convert it to its equivalent CSR sparsity pattern instead of the block sparsity pattern.

Reply:
The problem I see is that one would expect the size of the matrix and its sparsity pattern to be the same, which will not be the case here. But anyway, I guess we will postpone this discussion to a later point.